go/libraries/doltcore/table/untyped/csv/reader.go

```diff
@@ -455,3 +470,43 @@ func (csvr *CSVReader) parseQuotedField(rs *recordState) (kontinue bool, err err
 		}
 	}
 }
+
+// interpretRowSizeError returns a format map (written as a string) of a set of columns to their row values. It also
+// returns a slice of an unused strings.
+func interpretRowSizeError(schema schema.Schema, rowVals []*string) (string, []string) {
+	cols := schema.GetAllCols().GetColumns()
+
+	keyValPairs := make([][]string, len(cols))
+	unusedRowValues := make([]string, 0)
+
+	// 1. Start by adding all cols to the map and their relevant pair
+	for i, col := range cols {
+		if i >= len(rowVals) {
+			keyValPairs[i] = []string{col.Name, ""}
+		} else {
+			keyValPairs[i] = []string{col.Name, *rowVals[i]}
+		}
+	}
+
+	// 2. Append any unused row values to print to the user
+	for i := len(cols); i < len(rowVals); i++ {
+		if rowVals[i] == nil {
```

Comment:
(minor) do you need that Sprintf? Seems simpler just to use a literal value and not call Sprintf.

Reply:
%q quotes the string. felt helpful for a map output